### PR TITLE
Add documentation of the obsgendiff buildflag

### PR DESCRIPTION
**Pull request by dcermak**
~~@adrianschroeter Unfortunately I have no idea what this flag actually does~~

@adrianschroeter I've tried to document the flag to the best of my knowledge.

cc @Vogtinator

### CODE REVIEW

**Comment by Vogtinator** [comment]
All it actually does is: https://github.com/openSUSE/obs-build/blob/e9765d9a64b3c5055701ca9157d426d1ea3610a2/build-recipe#L210

So pretty much all the logic is in `release-compare` and so the documentation here is actually about that (specific) case.

**Comment by dcermak** [comment]
I've reworded the text to make this part clearer.

File: xml/obs_build_config.xml [hunk 1]
```diff
@@ -233,6 +233,32 @@ Keep: debbuild</screen>
         </para>
        </listitem>
       </varlistentry>
+      <varlistentry>
+        <term><parameter>obsgendiff:</parameter></term>
+       <listitem>
+        <para>
+          OBS can run a program to compare the result of the current build with
```
Comment (Vogtinator) [inline xml/obs_build_config.xml:240]: I don't think it actually does any comparison, it only extracts the data for external tools to be able to do so

File: xml/obs_build_config.xml [hunk 2]
```diff
@@ -233,6 +233,32 @@ Keep: debbuild</screen>
...
+          the previous result, e.g. to generate a difference or a changelog from
+          the diff.
+        </para>
+        <para>
+          OBS will run all scripts in
+          <filename>/usr/lib/build/obsgendiff.d/</filename> in the buildroot
```
Comment (Vogtinator) [inline xml/obs_build_config.xml:246]: Might be confused with the rpm `%buildroot`

File: xml/obs_build_config.xml [hunk 3]
```diff
@@ -233,6 +233,32 @@ Keep: debbuild</screen>
...
+          when this flag is set (which contains the previous build and current
```
Comment (Vogtinator) [inline xml/obs_build_config.xml:247]: I don't think it contains the previous build (ignoring the "oldpackages" for `build-compare`)

**Pull request merged**